docs/src/pages/guides/pickers-migration/pickers-migration.md

```diff
@@ -0,0 +1,118 @@
+
+This guide is an overview of the core concepts that were changed from the pickers v3.2.10.
+
+## ⬇️ Install
+
+First of all you need to install the `@material-ui/lab` if it's not installed. Nothing more required.
+
+## 🌚 Imports
+
+We do not expose the `keyboard` version of pickers anymore. All the version of mobile and desktop pickers implements keyboard inputs for accessibility reasons.
+
+```diff
+- import { KeyboardDatePicker } from '@material-ui/pickers'
++ import { DatePicker } from '@material-ui/lab'
+
+- <KeyboardDatePicker />
++ <DatePicker />
+```
+
+Also instead of `variant` prop specific wrappers of components were moved to different imports. So you will not bundle the `Dialog` if you are using only desktop picker view.
+
+- `<DesktopDatePicker />` – Only desktop view.
+- `<MobileDatePicker />` – Only mobile view.
+- `<DatePicker />` – Mobile or Desktop view according to the user **pointer** preference.
+- `<StaticDatePicker />` – The picker view itself, without input or any other wrapper.
+
+```diff
++ import { DesktopDatePicker } from '@material-ui/lab'
+
+- <DatePicker variant="inline" />
++ <DesktopDatePicker />
+```
+
+The same convention for `TimePicker` – `<DesktopTimePicker>` and `<MobileTimePicker />`.
+
+## MuiPickersUtilsProvider
+
+The `MuiPickersUtilsProvider` was removed in favor of `LocalizationProvider`. Also, pickers do not require to install date-io adapters manually. Everything comes together with the `lab`.
+
+❌ Before:
+
+```jsx
+import AdapterDateFns from '@date-io/date-fns';
+import { MuiPickersUtilsProvider } from '@material-ui/pickers/MuiPickersUtilsProvider';
+```
+
+✅ After:
+
+```jsx
+import AdapterDateFns from '@material-ui/lab/AdapterDateFns';
+import LocalizationProvider from '@material-ui/lab/LocalizationProvider';
+
+
+function App() {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      ...
+    </LocalizationProvider>
+  )
+);
+```
+
+## Render input
+
+We introduced a new **required** `renderInput` prop. It simplifies using different than material-ui text field input components.
+
+```jsx
+<DatePicker renderInput={(props) => <TextField {...props} />} />
+<TimePicker renderInput={(props) => <TextField {...props} />} />
+```
+
+Also before we were spreading props down to the `<TextField />` props. From now you need to use this new **required prop** `renderInput` to render the prop:
+
+```diff
+  <DatePicker
+-   label="Date"
+-   helperText="Something"
++   renderInput={props => <TextField label="Date" helperText="Something" /> }
+  />
+```
+
+## State management
+
+The state/value management logic for pickers was rewritten from scratch. They will now call the `onChange` prop when each view of picker ends its input and `onError` handler is now completely different. Triple-check your pickers with forms integration, because form-integration issues can be not obvious.
```

Comment:
```suggestion
The state/value management logic for pickers was rewritten from scratch. Pickers will now call the `onChange` prop when each view of picker ends its input. The `onError` handler is  also completely different. Triple-check your pickers with forms integration, because form-integration issues can be subtle.
```